code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,354 @@
+
+/// Callback for a repeating 10s timer to ensure both pylons are active (otherwise delete) and counts the number of marines groundside (would cause hatching to expedite).
+/obj/effect/alien/resin/king_cocoon/proc/check_pylons()
+	var/datum/hive_status/hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
+
+	if(length(hive.active_endgame_pylons) < 2)
+		qdel(src)
+		return
+
+	if(chosen_candidate || rolling_candidates)
+		return
+
+	if(icon_state == "hatching" || icon_state == "hatched")
+		return
+
+	var/groundside_humans = 0
+	for(var/mob/living/carbon/human/current_human as anything in GLOB.alive_human_list)
+		if(!(isspecieshuman(current_human) || isspeciessynth(current_human)))
+			continue
+
+		var/turf/turf = get_turf(current_human)
+		if(is_ground_level(turf?.z))
+			groundside_humans += 1
+			
+			if(groundside_humans > 12)
+				return
+
+	// Too few marines are now groundside, hatch immediately
+	deltimer(timer)
+	start_vote(expedite = TRUE)
+
+/// Causes the cocoon to change visually for growing and initiates the next timer.
+/obj/effect/alien/resin/king_cocoon/proc/start_growing()
+	icon_state = "growing"
+	timer = addtimer(CALLBACK(src, PROC_REF(announce_halfway)), 5 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+
+/// Causes the halfway announcements and initiates the next timer.
+/obj/effect/alien/resin/king_cocoon/proc/announce_halfway()
+	timer = addtimer(CALLBACK(src, PROC_REF(start_vote)), 4 MINUTES, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_DELETE_ME)
+
+	marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP DETECTED IN [uppertext(get_area_name(loc))].\n\nESTIMATED TIME UNTIL COMPLETION - 5 MINUTES.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
+	var/datum/hive_status/hive
+	for(var/cur_hive_num in GLOB.hive_datum)
+		hive = GLOB.hive_datum[cur_hive_num]
+		if(!length(hive.totalXenos))
+			continue
+		if(cur_hive_num == hive_number)
+			xeno_announcement(SPAN_XENOANNOUNCE("The King will hatch in approximately 5 minutes."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+		else
+			xeno_announcement(SPAN_XENOANNOUNCE("Another hive's King will hatch in approximately 5 minutes."), cur_hive_num, XENO_GENERAL_ANNOUNCE)
+
+#define KING_PLAYTIME_HOURS (50 HOURS)
+
+/**
+ * Returns TRUE is the candidate passed is valid: Returns TRUE is the candidate passed is valid: Has client, not facehugger, not lesser drone, not banished, and conditionally on playtime.
+ *
+ * Arguments:
+ * * hive: The hive_status to check banished ckeys against
+ * * candidate: The mob that we want to check
+ * * playtime_restricted: Determines whether being below KING_PLAYTIME_HOURS makes the candidate invalid
+ * * skip_playtime: Determines whether being above KING_PLAYTIME_HOURS makes the candidate invalid (does nothing unless playtime_restricted is FALSE)
+ */
+/obj/effect/alien/resin/king_cocoon/proc/is_candidate_valid(datum/hive_status/hive, mob/candidate, playtime_restricted = TRUE, skip_playtime = TRUE)
+	if(!candidate.client)
```

Comment:
```suggestion
	if(!candidate?.client)
```
Not gamebreaking, but this is now possible and we don't need to runtime when it happens.